File: app/types/matching.ts
export interface TableCandidate {
  source_schema: string;
  source_table: string;
  target_schema: string;
  target_table: string;
  confidence: number;
  match_reason: string;
  details: MatchDetails;
}

export interface MatchDetails {
  column_similarity: number;
  datatype_compatibility: number;
  key_match: number;
  fk_relationship_match: number;
  column_comparisons: ColumnComparison[];
}

export interface ColumnComparison {
  source_column: string;
  target_column: string | null;
  source_type: string;
  target_type: string | null;
  matched: boolean;
  match_type: 'exact' | 'name' | 'type' | 'none';
}

export interface MatchingConfig {
  exact_name_weight: number;
  suffix_pattern_weight: number;
  column_name_weight: number;
  datatype_weight: number;
  key_match_weight: number;
  fk_match_weight: number;
  auto_approve_threshold: number;
  min_confidence_threshold: number;
}

export const DEFAULT_MATCHING_CONFIG: MatchingConfig = {
  exact_name_weight: 1.0,
  suffix_pattern_weight: 0.95,
  column_name_weight: 0.4,
  datatype_weight: 0.3,
  key_match_weight: 0.2,
  fk_match_weight: 0.1,
  auto_approve_threshold: 90,
  min_confidence_threshold: 50,
};

File: app/types/matching.py
from dataclasses import dataclass
from typing import List, Optional


@dataclass
class ColumnComparison:
    source_column: str
    target_column: Optional[str]
    source_type: str
    target_type: Optional[str]
    matched: bool
    match_type: str


@dataclass
class MatchDetails:
    column_similarity: float
    datatype_compatibility: float
    key_match: float
    fk_relationship_match: float
    column_comparisons: List[ColumnComparison]


@dataclass
class TableCandidate:
    source_schema: str
    source_table: str
    target_schema: str
    target_table: str
    confidence: float
    match_reason: str
    details: MatchDetails


@dataclass
class MatchingConfig:
    exact_name_weight: float = 1.0
    suffix_pattern_weight: float = 0.95
    column_name_weight: float = 0.4
    datatype_weight: float = 0.3
    key_match_weight: float = 0.2
    fk_match_weight: float = 0.1
    auto_approve_threshold: float = 90.0
    min_confidence_threshold: float = 40.0


DEFAULT_MATCHING_CONFIG = MatchingConfig()
